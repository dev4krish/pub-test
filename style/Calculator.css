
    .flexColumn {
        display: flex;
        flex-direction: column;
    }
.flexRow {
    display: flex;
    flex-direction: row;
}
.button_pannel {
    flex:5;
    margin: 0px;
}
.input_pannel {
    display: flex;
    flex:1;
    margin:0px;
}
.d1 {
    width: 100%;
    height: 100%;
    display:flex;
    flex-direction: column;
    border: 1px groove aliceblue;  
    box-sizing: border-box;
    justify-content: center;
    margin: 0px;
    
}


.button {
    cursor: pointer;
    position: relative;
    padding: 0px;
    margin: 0px;
    box-shadow: inset 1px 1px 0 0 #e3e7e9;
    border: none;
    background-color:#F8F8FF;
    font-size: 12px;
    color: #000000;
    width :100%;
    height:2em;
    overflow: hidden;
}
.s1 { 
    flex: 1; 
}
.s2 { 
    flex: 2; 
}
.s3 { 
    flex: 3; 
}
.inputDiv {
    flex: 1;
    color: #46494d;
    overflow: auto;
    max-width: 100%;
    text-align: right;
    margin: 0px;
}
:focus {
    outline-width:0px;
    background-color: #FFFAF0;
}


