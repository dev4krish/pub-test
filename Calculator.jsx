import React from 'react';
import style from './style.css';

class Calculator extends React.Component{
    constructor(){
        super()
        this.handleInput=this.handleInput.bind(this);
        this.handleCalculation=this.handleCalculation.bind(this);
        this.clearInputBox=this.clearInputBox.bind(this);
        this.state={input:''};
    }
    handleInput(e){
        this.setState({input:e.target.value==''?'':this.state.input+e.target.value});
    }
    handleCalculation(e){
            this.setState({input:eval(this.state.input)});
    }
    clearInputBox(e){
        let inputValue=document.getElementById('input').value;
        let length=inputValue.length;
        inputValue=inputValue.substring(0,length-1);
        this.setState({input:inputValue});
        //String.fromCharCode
        
    }
    render(){
        const input=this.state.input;
        //<ShowCalculatedValue inputProps={input}> </ShowCalculatedValue>
        return(<fieldset className={style.outer}>
            <div className={style.innerDiv} >
                <input className={style.inputBox} id='input' value={input} onChange={this.handleInput} />
               
            </div>
            <div className={style.innerDiv}>
            <button className={style.button} value='0' onClick={this.handleInput}>0</button>
            <button className={style.button} value='1' onClick={this.handleInput}>1</button>
            <button className={style.button} value='2' onClick={this.handleInput}>2</button>
            </div>
            <div className={style.innerDiv}>
            <button className={style.button} value='3' onClick={this.handleInput}>3</button>
            <button className={style.button} value='4' onClick={this.handleInput}>4</button>
            <button className={style.button} alue='5' onClick={this.handleInput}>5</button>
            </div>
            <div className={style.innerDiv}>    
            <button className={style.button} value='6' onClick={this.handleInput}>6</button>
            <button className={style.button} value='7' onClick={this.handleInput}>7</button>
            <button className={style.button} value='8' onClick={this.handleInput}>8</button>
            </div>
            <div className={style.innerDiv}>
            <button className={style.button} value='9' onClick={this.handleInput}>9</button>
            <button className={style.button} value='+' onClick={this.handleInput}>+</button>
            <button className={style.button} value='-' onClick={this.handleInput}>-</button>
            </div>
            <div className={style.innerDiv}>
            <button className={style.button} value='*' onClick={this.handleInput}>*</button>
            <button className={style.button} value='/' onClick={this.handleInput}>/</button>
            <button className={style.button} value='%' onClick={this.handleInput}>%</button>
            </div>
            <div className={style.innerDiv}>
            <button className={style.button} value='' onClick={this.clearInputBox} >del</button>
            <button className={style.button} value='' onClick={this.handleInput} >Clear</button>
            <button className={style.button} onClick={this.handleCalculation} >=</button>
            </div>
        </fieldset>
        );
    }
}
class ShowCalculatedValue extends React.Component{
    render(){
        //return (this.props.inputProps==''?null:<div>{this.props.inputProps}</div>);
        return (<div>{this.props.inputProps}</div>)
    }
}

export default Calculator;