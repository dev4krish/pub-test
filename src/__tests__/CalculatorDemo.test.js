import React from 'react';
import Calculator from '../CalculatorDemo.jsx';
import renderer from 'react-test-renderer';

//dummy test to verify the jest setup
/*describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});*/


describe('<Calculator/>', () => {

  it('should render calculator', () => {
    const rendered = renderer.create(
      <Calculator roundOff="1" />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('should render calculator', () => {
    const rendered = renderer.create(
      <Calculator roundOff="none" />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});


