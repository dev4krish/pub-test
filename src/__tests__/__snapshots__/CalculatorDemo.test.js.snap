// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator/> should render calculator 1`] = `
<div
  className="undefined"
>
  <div
    className={undefined}
  >
    <input
      className="undefined"
      id="input"
      tabIndex="1"
      value=""
    />
  </div>
  <div
    className="undefined"
  >
    <div
      className="undefined"
    >
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="8"
          onClick={[Function]}
          tabIndex="2"
          value="back"
        >
          del
        </button>
        <button
          className="undefined"
          data-code="67"
          onClick={[Function]}
          tabIndex="3"
          value="c"
        >
          Clear
        </button>
        <button
          className="undefined"
          data-code="shift+187"
          onClick={[Function]}
          tabIndex="4"
          value="+"
        >
          +
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="49"
          onClick={[Function]}
          tabIndex="6"
          value="1"
        >
          1
        </button>
        <button
          className="undefined"
          data-code="50"
          onClick={[Function]}
          tabIndex="7"
          value="2"
        >
          2
        </button>
        <button
          className="undefined"
          data-code="51"
          onClick={[Function]}
          tabIndex="8"
          value="3"
        >
          3
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="52"
          onClick={[Function]}
          tabIndex="10"
          value="4"
        >
          4
        </button>
        <button
          className="undefined"
          data-code="53"
          onClick={[Function]}
          tabIndex="11"
          value="5"
        >
          5
        </button>
        <button
          className="undefined"
          data-code="54"
          onClick={[Function]}
          tabIndex="12"
          value="6"
        >
          6
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="55"
          onClick={[Function]}
          tabIndex="14"
          value="7"
        >
          7
        </button>
        <button
          className="undefined"
          data-code="56"
          onClick={[Function]}
          tabIndex="15"
          value="8"
        >
          8
        </button>
        <button
          className="undefined"
          data-code="57"
          onClick={[Function]}
          tabIndex="16"
          value="9"
        >
          9
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="48"
          onClick={[Function]}
          tabIndex="17"
          value="0"
        >
          0
        </button>
        <button
          className="undefined"
          data-code="190"
          onClick={[Function]}
          tabIndex="18"
          value="."
        >
          .
        </button>
      </div>
    </div>
    <div
      className="undefined"
    >
      <button
        className="undefined"
        data-code="189"
        onClick={[Function]}
        tabIndex="5"
        value="-"
      >
        -
      </button>
      <button
        className="undefined"
        data-code="shift+56"
        onClick={[Function]}
        tabIndex="9"
        value="*"
      >
        *
      </button>
      <button
        className="undefined"
        data-code="191"
        onClick={[Function]}
        tabIndex="13"
        value="/"
      >
        /
      </button>
      <button
        className="undefined"
        data-code="13"
        onClick={[Function]}
        tabIndex="19"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`<Calculator/> should render calculator 2`] = `
<div
  className="undefined"
>
  <div
    className={undefined}
  >
    <input
      className="undefined"
      id="input"
      tabIndex="1"
      value=""
    />
  </div>
  <div
    className="undefined"
  >
    <div
      className="undefined"
    >
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="8"
          onClick={[Function]}
          tabIndex="2"
          value="back"
        >
          del
        </button>
        <button
          className="undefined"
          data-code="67"
          onClick={[Function]}
          tabIndex="3"
          value="c"
        >
          Clear
        </button>
        <button
          className="undefined"
          data-code="shift+187"
          onClick={[Function]}
          tabIndex="4"
          value="+"
        >
          +
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="49"
          onClick={[Function]}
          tabIndex="6"
          value="1"
        >
          1
        </button>
        <button
          className="undefined"
          data-code="50"
          onClick={[Function]}
          tabIndex="7"
          value="2"
        >
          2
        </button>
        <button
          className="undefined"
          data-code="51"
          onClick={[Function]}
          tabIndex="8"
          value="3"
        >
          3
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="52"
          onClick={[Function]}
          tabIndex="10"
          value="4"
        >
          4
        </button>
        <button
          className="undefined"
          data-code="53"
          onClick={[Function]}
          tabIndex="11"
          value="5"
        >
          5
        </button>
        <button
          className="undefined"
          data-code="54"
          onClick={[Function]}
          tabIndex="12"
          value="6"
        >
          6
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="55"
          onClick={[Function]}
          tabIndex="14"
          value="7"
        >
          7
        </button>
        <button
          className="undefined"
          data-code="56"
          onClick={[Function]}
          tabIndex="15"
          value="8"
        >
          8
        </button>
        <button
          className="undefined"
          data-code="57"
          onClick={[Function]}
          tabIndex="16"
          value="9"
        >
          9
        </button>
      </div>
      <div
        className="undefined"
      >
        <button
          className="undefined"
          data-code="48"
          onClick={[Function]}
          tabIndex="17"
          value="0"
        >
          0
        </button>
        <button
          className="undefined"
          data-code="190"
          onClick={[Function]}
          tabIndex="18"
          value="."
        >
          .
        </button>
      </div>
    </div>
    <div
      className="undefined"
    >
      <button
        className="undefined"
        data-code="189"
        onClick={[Function]}
        tabIndex="5"
        value="-"
      >
        -
      </button>
      <button
        className="undefined"
        data-code="shift+56"
        onClick={[Function]}
        tabIndex="9"
        value="*"
      >
        *
      </button>
      <button
        className="undefined"
        data-code="191"
        onClick={[Function]}
        tabIndex="13"
        value="/"
      >
        /
      </button>
      <button
        className="undefined"
        data-code="13"
        onClick={[Function]}
        tabIndex="19"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
